File: the-method-of-programming/ch01-string/exercises/03.cpp
#include <cstring>
#include <algorithm>
#include <iostream>

void countOccurrences(const char *str) {
  const int len = strlen(str);
  int count[256];
  std::fill(count, count + 256, 0);

  for (int i = 0; i < len; ++i) {
    ++count[(int)str[i]];
  }

  for (int i = 0; i < 256; ++i) {
    if (count[i] > 0) {
      std::cout << "[" << (char)i << "]" << count[i] << " ";
    }
  }
  std::cout << std::endl;
}

int main() {
  countOccurrences("*Riophae*Lee*");

  return 0;
}

File: the-method-of-programming/ch01-string/exercises/03.js
function countOccurrences(str) {
  const count = {}
  for (let i = 0; i < str.length; ++i) {
    const char = str[i]
    count[char] = (count[char] || 0) + 1
  }
  console.log(count)
}

function main() {
  countOccurrences('*Riophae*Lee*')
}

main()
